Meadowlark-js/packages/meadowlark-core/src/handler/Query.ts

```diff
@@ -67,15 +67,14 @@ export async function query(frontendRequest: FrontendRequest): Promise<FrontendR
     return { statusCode: 500, headers: frontendRequest.middleware.headerMetadata };
   }
 
-  if (response === 'QUERY_FAILURE_INVALID_QUERY') {
+  if (response === 'QUERY_FAILURE_INVALID_QUERY' || response === 'QUERY_FAILURE_CONNECTION_ERROR') {
```

Comment:
QUERY_FAILURE_INVALID_QUERY needs to be a 500. It's a problem on Meadowlark's side, not OpenSearch's side.